src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -1038,5 +1066,219 @@ private async Task<RepositoryBranch> GetRepositoryBranch(string repo, string bra
+            if (update.SourceSha == codeFlowStatus.SourceSha)
+            {
+                _logger.LogInformation("PR {url} for {subscription} is up to date ({sha})",
+                    pr.Url,
+                    update.SubscriptionId,
+                    update.SourceSha);
+                return ActionResult.Create(false, "PR cannot be updated");
+            }
+
+            try
+            {
+                await _pcsClient.CodeFlow.FlowAsync(new CodeFlowRequest
+                {
+                    BuildId = update.BuildId,
+                    SubscriptionId = update.SubscriptionId,
+                    PrBranch = codeFlowStatus.PrBranch,
+                    PrUrl = pr.Url,
+                });
+
+                codeFlowStatus.SourceSha = update.SourceSha;
+
+                await _codeFlowState.StoreStateAsync(codeFlowStatus);
+                await _pullRequestState.StoreStateAsync(pr);
+                await _pullRequestCheckState.SetReminderAsync();
+
+                await _pullRequestUpdateState.RemoveStateAsync();
+                await _pullRequestUpdateState.UnsetReminderAsync();
+            }
+            catch (Exception e)
+            {
+                // TODO https://github.com/dotnet/arcade-services/issues/3318: Handle this
+                _logger.LogError(e, "Failed to request branch update for PR {url} for subscription {subscriptionId}",
+                    pr.Url,
+                    update.SubscriptionId);
+            }
+
+            return ActionResult.Create(true, "New changes requested from PCS");
+        }
+
+        private async Task<ActionResult<bool>> RequestCodeFlowBranchAsync(UpdateAssetsParameters update, string targetBranch)
+        {
+            CodeFlowStatus codeFlowUpdate = new()
+            {
+                PrBranch = GetNewBranchName(targetBranch),
+                SourceSha = update.SourceSha,
+            };
+
+            _logger.LogInformation(
+                "New code flow request for subscription {subscriptionId}. Requesting branch {branch} from PCS",
+                update.SubscriptionId,
+                codeFlowUpdate.PrBranch);
+
+            try
+            {
+                await _pcsClient.CodeFlow.FlowAsync(new CodeFlowRequest
+                {
+                    BuildId = update.BuildId,
+                    SubscriptionId = update.SubscriptionId,
+                    PrBranch = codeFlowUpdate.PrBranch,
+                });
+            }
+            catch (Exception e)
+            {
+                // TODO https://github.com/dotnet/arcade-services/issues/3318: Handle this
+                _logger.LogError(e, "Failed to request new branch {branch} for subscription {subscriptionId}",
+                    codeFlowUpdate.PrBranch,
+                    update.SubscriptionId);
+                return ActionResult.Create(false, $"Failed to call PCS when requesting branch {codeFlowUpdate.PrBranch}");
```

Comment:
Do we really need a string reason in the ActionResult? We have a log record just above, and I see no value in duplicating the error. We are not analyzing them, but just logging. I can understand having some additional information like `enum Reason { GitHubFailure, etc}` which you can analyze in the client code and do some follow-up code.
^^ general question for future consideration as we have this in multiple places.

but I'd start with small and extend ActionResult
```
public ActionResult(T result) {..} // without message

public string? Message { get; } // either nullable or default
```

Reply:
I don't think I will touch this as it is part of the core Service Fabric infrastructure of Maestro we have here. The `ActionResult` is then going into the SQL and stored.

I would say this all would go away if we transitioned from Maestro to PCS.